src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -83,10 +89,25 @@ export class RenderLineInput {
 				? RenderWhitespace.All
 				: renderWhitespace === 'boundary'
 					? RenderWhitespace.Boundary
-					: RenderWhitespace.None
+					: renderWhitespace === 'selection'
+						? RenderWhitespace.Selection
+						: RenderWhitespace.None
 		);
 		this.renderControlCharacters = renderControlCharacters;
 		this.fontLigatures = fontLigatures;
+		this.selectionsOnLine = selectionsOnLine;
+	}
+
+	private sameSelection(otherSelections?: Range[]): boolean {
```

Comment:
Same personal preference to `Range[] | null` (or `LineRange[] | null` once you add that...).